vllm/model_executor/layers/quantization/quant_llm.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+import torch.nn as nn
+
+from vllm import _custom_ops as ops
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.logger import init_logger
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.layers.quantization.utils.quant_llm_utils import (
+    _SPLIT_K_MAP, from_scaled_tc_fpx, to_scaled_tc_fpx)
+from vllm.model_executor.utils import set_weight_attrs
+
+logger = init_logger(__name__)
+
+
+class QuantLLMFPConfig(QuantizationConfig):
+    """Config for QuantLLM FP quantizer. It supports fp4,
+    fp5, fp6, fp7.
+    
+    Reference: https://arxiv.org/abs/2401.14112
+    
+    Args: 
+        weight_bits: the target quantization bits, should be one of
+            4, 5, 6, 7.
+    """
+
+    def __init__(
+        self,
+        weight_bits: int = 6,
+        exp_bits: int = 2,
+    ) -> None:
+        self.weight_bits = weight_bits
+        self.exponent_bits = exp_bits
+
+        self.mantissa_bits = weight_bits - self.exponent_bits - 1
+
+        self.valid_types = [torch.float16]
+
+        if self.weight_bits not in [4, 5, 6, 7]:
+            raise ValueError(
+                "Currently, only 4-bit, 5-bit, 6-bit, and 7-bit "
+                "quantization are supported for QuantLLM FP quantizaiton, "
+                f"but got {self.weight_bits} bits.")
+
+        if get_tensor_model_parallel_rank() == 0:
+            logger.info("Loading model in FP%s_E%sM%s format.",
+                        self.weight_bits, self.exponent_bits,
+                        self.mantissa_bits)
+
+    def __repr__(self) -> str:
+        return (f"QuantLLMFPConfig(weight_bits={self.weight_bits}), "
+                f"exponent_bits={self.exponent_bits}")
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "QuantLLMFP"
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "QuantLLMFPConfig":
+        weight_bits = cls.get_from_keys(config, ["bits"])
+        exp_bits = cls.get_from_keys(config, ["exp_bits"])
+        return cls(weight_bits=weight_bits, exp_bits=exp_bits)
+
+    def get_linear_method(self) -> "QuantLLMFPLinearMethod":
+        return QuantLLMFPLinearMethod(self)
+
+    def get_scaled_act_names(self) -> List[str]:
+        return []
+
+    @classmethod
+    def get_supported_act_dtypes(cls) -> List[torch.dtype]:
+        return [torch.half]
+
+    @classmethod
+    # Need to figure it out
+    def get_min_capability(cls) -> int:
+        return 80
+
+    @staticmethod
+    def get_config_filenames() -> List[str]:
+        return [
+            "quant_config.json",
+            "quantize_config.json",
```

Comment:
Do you actually support a model that has the config listed out in it's config.json? I thought this was solely a CLI arg quantization method